pkg/R/utils.R

```diff
@@ -164,11 +176,14 @@ reserialize <- function(rdd) {
   if (!inherits(rdd, "RDD")) {
     stop("Argument 'rdd' is not an RDD type.")
   }
-  if (rdd@env$serialized) {
-    return(rdd)
-  } else {
+  if (rdd@env$serializedMode != "byte") {
+    if (rdd@env$serializedMode == "row") {
+      rdd@env$serialzedMode == "byte"
```

Comment:
No, we don't. The default mode in `getJRDD` is `byte`, but that only gets used in deciding whether `StringRRDD` or `RRDD` gets created. Once the function has diverted away from string, the `byte` and `row` versions are the same since we can just pass `colNames` as an empty list and create a normal RRDD. As a result, the serialization type in `getJRDD` is actually defined by the incoming RDD to allow for `worker.R` to correctly differentiate between `byte` and `row`.

Example: https://github.com/cafreeman/SparkR-pkg/blob/sparkr-sql/pkg/R/RDD.R#L159